core/src/main/java/com/sequenceiq/cloudbreak/service/salt/PartialSaltStateUpdateService.java

```diff
@@ -0,0 +1,61 @@
+package com.sequenceiq.cloudbreak.service.salt;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.orchestrator.host.TelemetryOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.metadata.OrchestratorMetadata;
+import com.sequenceiq.cloudbreak.orchestrator.metadata.OrchestratorMetadataProvider;
+import com.sequenceiq.cloudbreak.util.CompressUtil;
+
+@Service
+public class PartialSaltStateUpdateService {
+    private static final String SALT_DIRECTORY_PREFIX = "/salt/";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PartialSaltStateUpdateService.class);
+
+    @Inject
+    private OrchestratorMetadataProvider orchestratorMetadataProvider;
+
+    @Inject
+    private CompressUtil compressUtil;
+
+    @Inject
+    private TelemetryOrchestrator stateUpdateService;
```

Comment:
if the methods used here is not telemetry specific, then extract them/move them into a parent class, because injecting a TelemetryOrchestrator here is not a proper solution, since this logic has nothing to do with telemetry

Reply:
Done.